node/src/Pos/Client/Txp/Balances.hs

```diff
@@ -53,17 +58,31 @@ type BalancesEnv ext ctx m =
     , MonadMask m
     , MonadTxpMem ext ctx m)
 
+-- TODO [CSM-407] Move it somewhere :/
+-- | Get active pending transactions which are *not* in blockchain yet.
+getRecentTxs :: WebWalletModeDB ctx m => m [WithHash Tx]
+getRecentTxs = tryTopsortTxs . mapMaybe extract <$> WS.getPendingTxs
+  where
+    extract PendingTx{..} = do
+        let PtxApplying th = _ptxCond
+        Just $ WithHash (_thTx th) _ptxTxId
+    tryTopsortTxs txs = fromMaybe txs $ topsortTxs identity txs
+
 getOwnUtxosDefault :: BalancesEnv ext ctx m => [Address] -> m Utxo
 getOwnUtxosDefault addrs = do
     let (redeemAddrs, commonAddrs) = partition isRedeemAddress addrs
 
-    updates <- getUtxoModifier
+    mempoolUpdates <- getUtxoModifier
+    recentTxs <- getRecentTxs
+    let applyUpdates = alaf Endo foldMap applyTxToUtxoWarilyPure recentTxs
```

Comment:
I strongly believe that there was a less confusing way to do it (e. g. _foldl'_?...)